src/services/review/ReviewRequestService.ts

```diff
@@ -439,36 +440,74 @@ export default class ReviewRequestService {
     // NOTE: commits from github are capped at 300.
     // This implies that there might possibly be some files
     // whose commit isn't being returned.
-    await Promise.all(
-      commits.map(async ({ commit, sha }) => {
-        const { userId } = fromGithubCommitMessage(commit.message)
-        const author = await this.users.findByPk(userId)
-        const lastChangedTime = new Date(commit.author.date).getTime()
-        mappings[sha] = {
-          author: author?.email || commit.author.name,
-          unixTime: lastChangedTime,
-        }
-      })
+
+    // commitAuthorIds are linked to the commits by array index
+    const commitAuthorIds = commits.map(
+      ({ commit }) => fromGithubCommitMessage(commit.message).userId
```

Comment:
could we filter out the possibly undefined `commitAuthorIds` so that we don't have to do a type cast below for `validAuthorIds`?

Reply:
Not at this step, unless we change the code more 🤔 

This is explained in the comment above:
```
// commitAuthorIds are linked to the commits by array index
```
This is done so when we iterate over the commits again further below, we have the userId of that commit without having to parse the commit message again. See line 468 below:
```
    const authorId = commitAuthorIds[idx]
```

An alternative might be to store some temporary structure for mappings by commit sha, but that also adds a bit of mess in the function.